sylvia-derive/src/message.rs

```diff
@@ -118,26 +146,46 @@ impl<'a> StructMessage<'a> {
         let generics = emit_bracketed_generics(generics);
         let unused_generics = emit_bracketed_generics(unused_generics);
 
+        let base_name = match msg_ty {
+            MsgType::Instantiate => quote! { base_instantiate },
+            MsgType::Migrate => quote! { base_migrate },
+            _ => unimplemented!(),
+        };
+
         #[cfg(not(tarpaulin_include))]
         {
             quote! {
+                // This is the InstantiateMsg for the contract
                 #[allow(clippy::derive_partial_eq_without_eq)]
                 #[derive(#sylvia ::serde::Serialize, #sylvia ::serde::Deserialize, Clone, Debug, PartialEq, #sylvia ::schemars::JsonSchema)]
                 #[serde(rename_all="snake_case")]
                 pub struct #name #generics {
+                    pub base: <#contract_msg_types as ::abstract_app::better_sdk::sdk::SylviaAbstractContract> #abstract_name,
+                    pub module: #app_msg_name #generics
+                }
+
+                // This is the InstantiateMsg for the app
+                #[allow(clippy::derive_partial_eq_without_eq)]
+                #[derive(#sylvia ::serde::Serialize, #sylvia ::serde::Deserialize, Clone, Debug, PartialEq, #sylvia ::schemars::JsonSchema)]
+                #[serde(rename_all="snake_case")]
+                pub struct #app_msg_name #generics {
                     #(pub #fields,)*
                 }
 
                 impl #generics #name #generics #where_clause {
-                    pub fn new(#(#parameters,)*) -> Self {
-                        Self { #(#fields_names,)* }
-                    }
 
                     pub fn dispatch #unused_generics(self, contract: &#contract_type, ctx: #ctx_type)
-                        #result #full_where
+                       -> #ret_type #full_where
                     {
-                        let Self { #(#fields_names,)* } = self;
-                        contract.#function_name(Into::into(ctx), #(#fields_names,)*).map_err(Into::into)
+
+                        let Self { base, module} = self;
+
+                        let #app_msg_name { #(#fields_names,)* } = module;
+                        let mut ctx = Into::into(ctx);
```

Comment:
The context allows for registering responses. It have to be common to both the base and the contract function